R4ML/R/zzz.R

```diff
@@ -85,286 +71,128 @@ auto_start_session <- function() {
 
   # we will use now the Java Spark Context
-  sysmlJavaSparkContext <- SparkR:::newJObject("org.apache.spark.api.java.JavaSparkContext",
-                        sysmlSparkContext)
-  assign("sysmlJavaSparkContext", sysmlJavaSparkContext, .GlobalEnv)
-  
+  r4ml.env$sysmlJavaSparkContext <- SparkR::sparkR.newJObject("org.apache.spark.api.java.JavaSparkContext", r4ml.env$sysmlSparkContext)
+
   # since sysmlSqlContext is used for most of the systemML related
   # code we need to have it.
-  sysmlSqlContext <- SparkR:::callJMethod(sc, "sqlContext")
-  assign("sysmlSqlContext", sysmlSqlContext, .GlobalEnv)
+  r4ml.env$sysmlSqlContext <- SparkR::sparkR.callJMethod(r4ml.env$sc, "sqlContext")
   
   # java logger and level settings
-  jlogger <- log4j.Logger$new("org")
-  assign("jlogger", jlogger, .GlobalEnv)
-  jlogger$setLevel("WARN")
+  r4ml.env$jlogger <- log4j.Logger$new("org")
+  r4ml.env$jlogger$setLevel("WARN")
 
   # static class sysml.RDDUtils
   # since actual class is static, we need this
-  sysml.RDDUtils <- sysml.RDDConverterUtils$new()
-  assign("sysml.RDDUtils", sysml.RDDUtils, .GlobalEnv)
-}
+  r4ml.env$sysml.RDDUtils <- sysml.RDDConverterUtils$new()
 
-# systemML related stopping i.e de-initialization
-sysml.stop <- function() {
-  if("jlogger" %in% ls(.GlobalEnv)) { rm(jlogger, envir = .GlobalEnv) }
-  if("sysml.RDDUtils" %in% ls(.GlobalEnv)) { rm(sysml.RDDUtils, envir = .GlobalEnv) }
-  if("sysmlSparkContext" %in% ls(.GlobalEnv)) { rm(sysmlSparkContext, envir = .GlobalEnv) }
-  if("sysmlJavaSparkContext" %in% ls(.GlobalEnv)) { rm(sysmlJavaSparkContext, envir = .GlobalEnv) }
-  if("sysmlSqlContext" %in% ls(.GlobalEnv)) { rm(sysmlSqlContext, envir = .GlobalEnv) }
-}
-
-# R4ML related initialization
-r4ml.init <- function() {
-  logsrc <- r4ml.env$PACKAGE_NAME
   # R4ML logger
   logger <- Logging$new()
-  assign("r4ml.logger", logger, .GlobalEnv)
+  r4ml.env$r4ml.logger <- logger
   loglevel <- r4ml.env$DEFAULT_LOG_LEVEL
   env_loglevel <- Sys.getenv("R4ML_LOG_LEVEL")
   if (nchar(env_loglevel) != 0) {
     loglevel <- base::toupper(env_loglevel)
-    r4ml.info(logsrc, 
-      sprintf("Changing the log level to %s as per env 'R4ML_LOG_LEVEL'", loglevel))
+    r4ml.info(logsrc, sprintf("Changing the log level to %s as per env 'R4ML_LOG_LEVEL'", loglevel))
   }
   logger$setLevel(loglevel)
-  
+
   # create the r4ml.fs
   fs <- create.r4ml.fs()
-  assign("r4ml.fs", fs, .GlobalEnv)
-  
-  # assign the current sparkr.env to the r4ml.env
-  assign("sparkr.env", SparkR:::.sparkREnv, r4ml.env)
+  r4ml.env$r4ml.fs <- fs
 
   # post startup message
-  r4ml.post.startup.message()
+  loglevel <- r4ml.env$DEFAULT_LOG_LEVEL
+  packageStartupMessage(sprintf("[%s]: Default log level will be set to '%s'", logsrc, loglevel))
+  packageStartupMessage(sprintf("[%s]: To change call r4ml.setLogLevel(NewLevel)", logsrc))
 
   # mark the session exists flag
   r4ml.env$R4ML_SESSION_EXISTS <- TRUE
-  
-}
-
-r4ml.end <- function() {
```

Comment:
I removed almost 200 lines of unnecessary code from this file and these one job functions became unnecessary.